fhir_zulip_nlp/fhir_zulip_nlp.py

```diff
@@ -1,143 +1,208 @@
 """FHIR Zulip NLP Analysis
-# TODO: turn into GH issue, link that doc there, and link GH issue here
-# TODO: set up env and put instructions in readme
+
+TODO's
+  1. turn into GH issue, link that doc there, and link GH issue here
+  2. set up env and put instructions in readme
+  3. Need to show a report of all queries where keyword returned no messages
+
 Resources:
   1. Project requirements: https://github.com/jhu-bids/fhir-zulip-nlp-analysis/issues/1
   2. Zulip API docs: https://zulip.com/api/rest
     - Links to specific endpoints/articles are included in functions below.
+    - Get all streams (probably not needed): https://zulip.com/api/get-streams
+    - https://zulip.com/api/get-messages
   3. The Zulip chat we're querying: https://chat.fhir.org/#
 
 Possible areas of improvement
   1. Save to calling directory, not project directory.
 """
+import json
 import os
-from typing import Dict, List
+import sys
+from typing import Dict, List, Union
 
 import pandas as pd
+import time
 import zulip
 from datetime import datetime
 
 # Vars
 PKG_DIR = os.path.dirname(os.path.realpath(__file__))
 PROJECT_DIR = os.path.join(PKG_DIR, '..')
+CACHE_DIR = os.path.join(PKG_DIR, 'cache')
+RAW_RESULTS_FILENAME = 'zulip_raw_results.csv'
 CONFIG = {
     # 'outdir': PROJECT_DIR,
     # 'report1_filename': 'fhir-zulip-nlp - frequencies and date ranges.csv',
     # 'report2_filename': 'fhir-zulip-nlp - age and date analyses.csv',
     'zuliprc_path': os.path.join(PROJECT_DIR, 'zuliprc'),  # rc = "runtime config"
-    'outpath': os.path.join(PROJECT_DIR, 'zulip_report.csv')
+    'chat_stream_name': 'terminology',
+    'num_messages_per_query': 1000,
+    'outpath_report1': os.path.join(PROJECT_DIR, 'zulip_report1.csv'),
+    'outpath_raw_results': os.path.join(PROJECT_DIR, RAW_RESULTS_FILENAME),
+    'cache_outpath': os.path.join(CACHE_DIR, RAW_RESULTS_FILENAME),
+}
+# TODO: need to account for spelling variations
+# TODO: need to account for overlap. CDA is a subset of C-CDA, so need to prune results for these miscatches.
```

Comment:
@rohaher I realized that there are some edge cases we have to worry about. added some TODOs to remind us